src/service/xhr-impl.js

```diff
@@ -106,6 +113,15 @@ export class Xhr {
    * @private
    */
   fetch_(input, init) {
+    if (this.ampdoc_) {
+      // TODO(@jridgewell, #9048): We can alternatively _always_ wait on
+      // #whenFirstVisible.
+      if (Math.random() < .01 && !viewerForDoc(this.ampdoc_).hasBeenVisible()) {
```

Comment:
Eventually, I believe, we'll use this condition and throw an actual error `==` deny the request in prerender with different origin.